libs/vkd3d/workgraphs.c

```diff
@@ -0,0 +1,3467 @@
+}
+
+static CONST_VTBL struct ID3D12StateObjectVtbl d3d12_state_object_vtbl =
+{
+    /* IUnknown methods */
+    d3d12_state_object_QueryInterface,
+    d3d12_state_object_AddRef,
+    d3d12_state_object_Release,
+    /* ID3D12Object methods */
+    d3d12_state_object_GetPrivateData,
+    d3d12_state_object_SetPrivateData,
+    d3d12_state_object_SetPrivateDataInterface,
+    (void *)d3d12_object_SetName,
+    /* ID3D12DeviceChild methods */
+    d3d12_state_object_GetDevice,
+};
+
+/* These only apply to RT PSOs. */
+static void * STDMETHODCALLTYPE d3d12_state_object_properties_GetShaderIdentifier(d3d12_state_object_properties_iface *iface,
+        LPCWSTR export_name)
+{
+    TRACE("iface %p, export_name %s.\n", iface, debugstr_w(export_name));
+    return NULL;
+}
+
+static UINT64 STDMETHODCALLTYPE d3d12_state_object_properties_GetShaderStackSize(d3d12_state_object_properties_iface *iface,
+        LPCWSTR export_name)
+{
+    TRACE("iface %p, export_name %s.\n", iface, debugstr_w(export_name));
+    return UINT32_MAX;
+}
+
+static UINT64 STDMETHODCALLTYPE d3d12_state_object_properties_GetPipelineStackSize(d3d12_state_object_properties_iface *iface)
+{
+    TRACE("iface %p\n", iface);
+    return UINT32_MAX;
+}
+
+static void STDMETHODCALLTYPE d3d12_state_object_properties_SetPipelineStackSize(d3d12_state_object_properties_iface *iface,
+        UINT64 stack_size_in_bytes)
+{
+    TRACE("iface %p, stack_size_in_bytes %llu!\n", iface, (unsigned long long)stack_size_in_bytes);
+}
+
+static D3D12_PROGRAM_IDENTIFIER * STDMETHODCALLTYPE d3d12_state_object_properties_GetProgramIdentifier(
+        d3d12_state_object_properties_iface *iface,
+        D3D12_PROGRAM_IDENTIFIER *ret,
+        LPCWSTR pProgramName)
+{
+    struct d3d12_wg_state_object *state_object = impl_from_ID3D12StateObjectProperties(iface);
+    size_t i;
+
+    TRACE("iface %p, ret %p, pProgramName %s!\n", iface, ret, debugstr_w(pProgramName));
+
+    memset(ret, 0, sizeof(*ret));
+    for (i = 0; i < state_object->programs_count; i++)
+    {
+        if (vkd3d_export_strequal(state_object->programs[i].name, pProgramName))
+        {
+            ret->OpaqueData[0] = i;
+            ret->OpaqueData[1] = (UINT64)(uintptr_t)state_object; /* Allows sanity checking. */
+            break;
+        }
+    }
+
+    return ret;
+}
+
+static CONST_VTBL struct ID3D12StateObjectProperties1Vtbl d3d12_state_object_properties_vtbl =
+{
+    /* IUnknown methods */
+    d3d12_state_object_properties_QueryInterface,
+    d3d12_state_object_properties_AddRef,
+    d3d12_state_object_properties_Release,
+    /* ID3D12StateObjectProperties methods */
+    d3d12_state_object_properties_GetShaderIdentifier,
+    d3d12_state_object_properties_GetShaderStackSize,
+    d3d12_state_object_properties_GetPipelineStackSize,
+    d3d12_state_object_properties_SetPipelineStackSize,
+    /* ID3D12StateObjectProperties1 methods */
+    d3d12_state_object_properties_GetProgramIdentifier,
+};
```

Comment:
bit of a nit perhaps, but given that these are different from the RT state_object_properties, can we rename these functions to something like `d3d12_wg_state_object_properties_*` instead to keep naming unique?